tests/input.rs

```diff
@@ -0,0 +1,226 @@
+use std::{
+    collections::VecDeque,
+    sync::{Arc, Mutex},
+};
+
+use bevy::{
+    ecs::event::{Event, Events},
+    input::InputPlugin,
+    input::{keyboard::KeyboardInput, ButtonState},
+    prelude::App,
+    prelude::{EventReader, KeyCode},
+    window::{ReceivedCharacter, WindowId, WindowPlugin},
+};
+
+#[test]
+fn test_input_received_character() {
+    let inputs = Arc::new(Mutex::new(VecDeque::from(['a', 'B'])));
+    let inputs2 = inputs.clone();
+    let system = move |mut chat_input_events: EventReader<ReceivedCharacter>| {
```

Comment:
The closure system is clever but makes this harder to read. I would just use a standard `fn` system.